### PR TITLE
Maximum value quorum for BroadcastReader

### PR DESCRIPTION
**Pull request by Termina1**
Fixes problem when we ask amount of transactions for user including mempool. Unfortunately, it's not always possible to propagate transaction to all nodes, so we need to ask all nodes and select maximum.

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/quorum/BroadcastQuorum.kt [hunk 1]
```diff
@@ -54,12 +47,12 @@ open class BroadcastQuorum(
         signature: ResponseSigner.Signature?,
         upstream: Upstream
     ) {
-        calls++
         if (txid == null && responseValue != null) {
             txid = responseValue
             sig = signature
             result = response
         }
+        resolvers.add(upstream)
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/quorum/BroadcastQuorum.kt:55]: It's done in `ValueAwareQuorum`

File: src/main/kotlin/io/emeraldpay/dshackle/quorum/BroadcastQuorum.kt [hunk 2]
```diff
@@ -68,16 +61,9 @@ open class BroadcastQuorum(
         signature: ResponseSigner.Signature?,
         upstream: Upstream
     ) {
-        // can be "message: known transaction: TXID", "Transaction with the same hash was already imported" or "message: Nonce too low"
-        calls++
-        if (result == null) {
-            result = response
-            sig = signature
-        }
-        resolvers.add(upstream)
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/quorum/BroadcastQuorum.kt:77]: Let's `resolvers.add(upstream)` in parent class then

File: src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package io.emeraldpay.dshackle.quorum
+
+import io.emeraldpay.dshackle.upstream.Upstream
+import io.emeraldpay.dshackle.upstream.signature.ResponseSigner
+import io.emeraldpay.etherjar.hex.HexQuantity
+import java.util.concurrent.atomic.AtomicReference
+
+class MaximumValueQuorum : CallQuorum, ValueAwareQuorum<String>(String::class.java) {
+    private val max: AtomicReference<Long?> = AtomicReference()
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt:9]: I think we can use just `Long`, there is no any parallel work

File: src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        upstream: Upstream
+    ) {
+        val value = responseValue?.let { str ->
+            HexQuantity.from(str).value.toLong()
+        }
+        if (value != null) {
+            max.getAndUpdate {
+                if (it == null || it < value) {
+                    sig = signature
+                    resolvers.clear()
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt:41]: Why `clear`?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 5]
```diff
@@ -196,7 +197,7 @@ class DefaultEthereumMethods(
             possibleNotIndexedMethods.contains(method) -> NotNullQuorum()
             specialMethods.contains(method) -> {
                 when (method) {
-                    "eth_getTransactionCount" -> AlwaysQuorum()
+                    "eth_getTransactionCount" -> MaximumValueQuorum()
                     "eth_getBalance" -> AlwaysQuorum()
                     "eth_blockNumber" -> NotLaggingQuorum(0)
                     else -> AlwaysQuorum()
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:203]: Also it's necessary to add again `eth_sendRawTransaction` here with `BroadcastQuorum` cause I removed it due to old logic in `BroadcastReader`

File: src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt (hunk 3 above)
Comment (Termina1) [reply src/main/kotlin/io/emeraldpay/dshackle/quorum/MaximumValueQuorum.kt:9]: in broadcast reader its parallel

**Review by KirillPamPam — APPROVED** [review]

**Pull request merged**